### PR TITLE
Update controller-runtime and controller-gen

### PR DESCRIPTION
**Pull request by fasmat**
## What problem does this PR solve?

This PR updates the k8s controller runtime to the most recent version as well as `controller-gen`.

## What's changed and how it works?

Dependencies were updated, depreciated features were migrated code was regenerated.

## Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`

## Cherry-pick to release branches (optional)

> This PR should be cherry-picked to the following release branches:

- [ ] release-2.6
- [ ] release-2.5

### CHANGELOG

> Must include at least one of them.

- [x] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

### Tests

> Must include at least one of them.

- [x] Unit test
- [x] E2E test
- [ ] Manual test

### Side effects

- [ ] **Breaking backward compatibility**

## DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by fasmat** [comment]
This PR probably requires the tag `rebuild-dev-env-image`

**Comment by fasmat** [comment]
@g1eny0ung can you add the label `rebuild-dev-env-image` to this PR? That should fix the failing CI steps. 🙂

**Comment by g1eny0ung** [comment]
> @g1eny0ung can you add the labels `rebuild-build-env-image` and `rebuild-dev-env-image` to this PR? That should fix the failing CI steps. 🙂

Got it. I have added them.

File: images/dev-env/Dockerfile [hunk 1]
```diff
@@ -5,12 +5,8 @@ FROM golang:1.20 AS go_install
 ENV GOPATH /go
 ENV CGO_ENABLED 0
 
-RUN mkdir -p /go/src/github.com/YangKeao/tools
-WORKDIR /go/src/github.com/YangKeao/tools
-RUN git clone https://github.com/YangKeao/tools.git -b v0.1.9-with-combine --depth 1 .
-RUN go install ./cmd/goimports
-
-RUN go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.1
+RUN go install golang.org/x/tools/cmd/goimports@v0.13.0
```
Comment (g1eny0ung) [inline images/dev-env/Dockerfile:8]: Nice change!

File: Makefile [hunk 2]
```diff
@@ -90,8 +90,8 @@ help: ## Display this help
 config: SHELL:=$(RUN_IN_DEV_SHELL)
 config: images/dev-env/.dockerbuilt ## Generate CRD manifests with controller-gen
 	cd ./api ;\
-		controller-gen crd:trivialVersions=true,preserveUnknownFields=false,crdVersions=v1 rbac:roleName=manager-role paths="./..." output:crd:artifacts:config=../config/crd/bases ;\
-		controller-gen crd:trivialVersions=true,preserveUnknownFields=false,crdVersions=v1 rbac:roleName=manager-role paths="./..." output:crd:artifacts:config=../helm/chaos-mesh/crds ;
```
Comment (g1eny0ung) [inline Makefile:94]: Just a note: `crd:trivialVersions=true,preserveUnknownFields=false` is not working when the default CRDs are `v1`.

**Review by g1eny0ung — APPROVED** [review]
Great job, @fasmat! Thank you for your hard work. The upgrade of controller-gen significantly optimizes the generated custom resource definitions.

I think this pull request is satisfactory. Please conduct an extra review at your convenience. @STRRL

File: Makefile (hunk 2 above)
Comment (fasmat) [reply Makefile:94]: I checked the source code of `controller-gen` and `trivialVersions` and `preserveUnknownFields` were removed as  options in this PR https://github.com/kubernetes-sigs/controller-tools/pull/607. The first release to include the change is https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.7.0 released ~ 2 years ago.

Comment (g1eny0ung) [reply Makefile:94]: Yes, I learned about it through this PR as well. But I didn't find anything about it on the kubebuilder documentation, so I'm leaving a note here.

**Comment by STRRL** [comment]
oh @fasmat you are awsome! I would take a look on this PR later!

**Pull request merged**